decentraland/unity-renderer
# fix: add AB test variant support for feature flags

## What does this PR change?

To properly support AB testing, we need feature flag variants to distinguish which experiment is used.
- fixed JSON serialization of `FeatureFlagVariant` class
- added support for defining variant name in featureFlag name, i.e `test_feature_flag_variant:variantName`
- updated feature flag name with `_variant` suffix, that are used in A/B test to support format needed for the analytics pipeline

[Kernel PR](https://github.com/decentraland/kernel/pull/461)
More info here and how to setup in Unleash: https://github.com/decentraland/adr/blob/main/docs/ADR-73-ab-testing.md

## How to test the changes?

Use this link to test changes https://play.decentraland.zone/?renderer-branch=fix/add-ab-test-variant-feature-flag?kernel-branch=fix%2Ffeature-flag-variant-parsing
Test with additional params 
- ENABLED_UI_BLUR_VARIANT, DISABLED_UI_BLUR_VARIANT
- ENABLED_USE_GPU_ONLY_MESH_VARIANT, DISABLED_USE_GPU_ONLY_MESH_VARIANT
  -  no visible change should be seen

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
